openml-h2o/src/main/java/com/feedzai/openml/h2o/algos/AbstractUnsupervisedH2OParamUtils.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 Feedzai
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.feedzai.openml.h2o.algos;
+
+import com.feedzai.openml.data.schema.DatasetSchema;
+import water.api.schemas3.FrameV3;
+import water.api.schemas3.KeyV3;
+import water.api.schemas3.ModelParametersSchemaV3;
+import water.fvec.Frame;
+
+import java.util.Map;
+
+/**
+ * Abstract class to parse H2O supervised algorithm params.
+ *
+ * @param <T> The concrete type of {@link ModelParametersSchemaV3 algorithm params}.
+ * @author Joao Sousa (joao.sousa@feedzai.com)
+ * @since 1.0.0
+ */
+public abstract class AbstractUnsupervisedH2OParamUtils<T extends ModelParametersSchemaV3> extends AbstractH2OParamUtils<T> {
```

Comment:
this class is very similar to AbstractSupervisedH2OParamUtils. is it possible to merge both classes?
like for example instead of passing the targetIndex pass the dataschema on the supervised methods

Reply:
that  would also simplify the H2OApp#trainSupervised and H2O#trainAnomalyDetection, allowing to merge both logics in the same methods

Reply:
Very nice tip! There's some nuance here though: the way params are generate in supervised is slightly different than in unsupervised due to the target field; in supervised schema is required (fail if there is no schema) while in unsupervised the schema is optional (if it exists add it, if not keep going).

Still, thanks to this I did some major refactors which made the code much simpler 👍